aiohttp_retry/client.py

```diff
@@ -80,13 +80,18 @@ def __init__(
 
         self._response: Optional[ClientResponse] = None
 
-    def _is_status_code_ok(self, code: int) -> bool:
-        if code >= 500 and self._retry_options.retry_all_server_errors:
-            return False
-        return code not in self._retry_options.statuses
+    async def _skip_retry(self, response: ClientResponse) -> bool:
+        skip_retry = False
+        if response.status >= 500 and self._retry_options.retry_all_server_errors:
+            return skip_retry
+        elif response.status not in self._retry_options.statuses:
+            skip_retry = (self._retry_options.evaluate_response_callback is None
+                          or await self._retry_options.evaluate_response_callback(response))
+        return skip_retry
```

Comment:
I love the early return idea. I would suggest to rewrite this code into the form:
```python
if response.status >= 500 and self._retry_options.retry_all_server_errors:
    return False

if response.status in self._retry_options.status:
    return False

if self._retry_options.evaluate_response_callback is None:
    return True

return await self._retry_options.evaluate_response_callback(response)
```